### PR TITLE
Incremental smt2 backend documentation

### PR DESCRIPTION
**Pull request by esteffin**
User documentation for the incremental smt2 backend.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by tautschnig** [comment]
How about added the newly proposed contents to `src/solvers/README.md` instead of creating a new file?

**Comment by esteffin** [comment]
> `README.md` instead of creating a new

It can be done.
Personally I prefer not adding it there as the added documentation is user-facing and not implementation specific as the one in `src/solvers/README.md`.

**Comment by tautschnig** [comment]
> > `README.md` instead of creating a new
> 
> It can be done. Personally I prefer not adding it there as the added documentation is user-facing and not implementation specific as the one in `src/solvers/README.md`.

Ok, thank you for the push-back. I don't really mind having it in that separate file, but then please do make sure your newly added documentation ends up being rendered as part of the doxygen-generated documentation. There is no point in having user-facing documentation stowed away in the source tree.

File: doc/cprover-manual/smt2-incr.md
Comment (TGWDB) [inline doc/cprover-manual/smt2-incr.md]: increase -> increases

File: doc/cprover-manual/smt2-incr.md
Comment (TGWDB) [inline doc/cprover-manual/smt2-incr.md]: allowing also -> also allowing

File: doc/cprover-manual/smt2-incr.md
Comment (TGWDB) [inline doc/cprover-manual/smt2-incr.md]: Also the -> Further, the

File: doc/cprover-manual/smt2-incr.md
Comment (TGWDB) [inline doc/cprover-manual/smt2-incr.md]: bit-vectors and uninterpreted -> bit-vectors, uninterpreted

File: doc/cprover-manual/smt2-incr.md
Comment (TGWDB) [inline doc/cprover-manual/smt2-incr.md]: No trace in the command here, remove "as well as the trace"

**Review by TGWDB — APPROVED** [review]
Minor text fixes only, otherwise approved.

File: doc/cprover-manual/index.md
Comment (tautschnig) [inline doc/cprover-manual/index.md:37]: Leaving it up to @kroening to make a call here, for he maintains the rendered version of this documentation. My own view is that we should not discuss a particular solver as part of this manual.

File: doc/cprover-manual/smt2-incr.md
Comment (esteffin) [reply doc/cprover-manual/smt2-incr.md]: Changed

File: doc/cprover-manual/smt2-incr.md
Comment (esteffin) [reply doc/cprover-manual/smt2-incr.md]: Changed

File: doc/cprover-manual/smt2-incr.md
Comment (esteffin) [reply doc/cprover-manual/smt2-incr.md]: changed

File: doc/cprover-manual/smt2-incr.md
Comment (esteffin) [reply doc/cprover-manual/smt2-incr.md]: changed

File: doc/cprover-manual/smt2-incr.md
Comment (esteffin) [reply doc/cprover-manual/smt2-incr.md]: Right, fixed

File: doc/cprover-manual/index.md
Comment (jimgrundy) [reply doc/cprover-manual/index.md:37]: I consider the "official" documentation now to be automatically rendered here: https://diffblue.github.io/cbmc/ Is that not the case? Will that not be the case for this file?

Where would you propose we discuss things specific to a particular solver?

**Review by jimgrundy — APPROVED** [review]

File: doc/cprover-manual/index.md
Comment (tautschnig) [reply doc/cprover-manual/index.md:37]: I agree that this one (github.io) is the right place; it will _also_ include the CPROVER manual (which is otherwise rendered on cprover.org), but isn't limited to that manual.

Comment (thomasspriggs) [reply doc/cprover-manual/index.md:37]: > I consider the "official" documentation now to be automatically rendered here: https://diffblue.github.io/cbmc/ Is that not the case? Will that not be the case for this file?

I can confirm that if I checkout and build the documentation locally as will be done on the documentation publishing job, then this new documentation will be included. Therefore it should be uploaded to the github.io page by github actions after this PR is merged.

The links to follow to find it will be -

1. User Guide in the side bar.
2. The link to "A short manual" in that pages's content.
3. The new page link will be under item 10.

> Where would you propose we discuss things specific to a particular solver?

I think this is the right place to add this documentation. Most information about the SAT solvers for instance wouldn't belong here, because they are compile/link time swappable not run-time swappable. However I think this list started by Enrico could be the right place to add documentation of the the dimacs file and external sat solver support, when someone takes the time to write it.

**Review by thomasspriggs — APPROVED** [review]

**Pull request merged**
